File: src/main/java/moe/giga/discord/permissions/AccessLevel.java
package moe.giga.discord.permissions;

public enum AccessLevel {
    USER, MOD, ADMIN, ROOT;

    public static AccessLevel fromString(String str) {
        switch (str.toLowerCase()) {
            case "user":
                return USER;
            case "mod":
                return MOD;
            case "admin":
                return ADMIN;
            case "root":
                return ROOT;
        }
        return USER; // return user by default
    }

    @Override
    public String toString() {
        switch (this) {
            case USER:
                return "user";
            case MOD:
                return "mod";
            case ADMIN:
                return "admin";
            case ROOT:
                return "root";
        }
        return "(unknown)";
    }

}

File: src/main/kotlin/moe/giga/discord/util/AccessLevel.kt
package moe.giga.discord.util

enum class AccessLevel {
    USER, MOD, ADMIN, ROOT;

    override fun toString(): String = when (this) {
        USER -> "user"
        MOD -> "mod"
        ADMIN -> "admin"
        ROOT -> "root"
    }

    companion object {

        fun fromString(str: String): AccessLevel = when (str.toLowerCase()) {
            "user" -> USER
            "mod" -> MOD
            "admin" -> ADMIN
            "root" -> ROOT
            else -> USER
        }
    }

}
